rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -569,6 +594,7 @@ void RSocketStateMachine::handleStreamFrame(
           std::move(framePayload.payload_),
           framePayload.header_.flagsComplete(),
           framePayload.header_.flagsNext());
+      resumeManager_->decrConsumerAllowance(streamId, 1);
```

Comment:
I think there is a potential bug here, since the framePayload might actually not have any payload, just the complete flag.
The stream state machine should be the only thing which knows what is the actual allowance.